subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/GradleEnterprisePluginExecutionPhaseStartedCallbackIntegrationTest.groovy

```diff
@@ -0,0 +1,197 @@
+            gradlePlugin {
+                plugins.create("javaProjectPlugin") {
+                    id = "project-plugin"
+                    implementationClass = "ProjectPlugin"
+                }
+            }
+        """
+        includedPlugin.file("src/main/groovy/precompiled-project-plugin.gradle") << """
+            println "precompiled plugin"
+        """
+        includedPlugin.file("src/main/java/ProjectPlugin.java") << """
+            import ${Plugin.name};
+            import ${Project.name};
+
+            public class ProjectPlugin implements Plugin<Project> {
+                @Override public void apply(Project p) {
+                    System.out.println("Project plugin");
+                }
+            }
+        """
+
+        def includedSettingsPlugin = testDirectory.createDir("included-settings-plugin")
+        includedSettingsPlugin.file("build.gradle") << """
+            ${applyPrecompiledGroovyPlugin()}
+            gradlePlugin {
+                plugins.create("javaSettingsPlugin") {
+                    id = "settings-plugin"
+                    implementationClass = "SettingsPlugin"
+                }
+            }
+        """
+        includedSettingsPlugin.file("src/main/java/SettingsPlugin.java") << """
+            import ${Plugin.name};
+            import ${Settings.name};
+
+            public class SettingsPlugin implements Plugin<Settings> {
+                @Override public void apply(Settings s) {
+                    System.out.println("Settings plugin");
+                }
+            }
+        """
+        includedSettingsPlugin.file("src/main/groovy/precompiled-settings-plugin.settings.gradle") << """
+            println "precompiled settings plugin"
+        """
+
+        def includedProject = testDirectory.createDir("included-project")
+        includedProject.file("build.gradle") << """
+            tasks.register("includedTask")
+        """
+        settingsFile << """
+            pluginManagement {
+                repositories {
+                    ${plugin.pluginRepository()}
+                }
+                includeBuild("${includedSettingsPlugin.name}")
+            }
+            plugins {
+                ${plugin.pluginDependency()}
+                id "settings-plugin"
+                id "precompiled-settings-plugin"
+            }
+
+            includeBuild("$includedPlugin.name")
+            includeBuild("$includedProject.name")
+        """
+
+        buildFile << """
+            plugins {
+                id "precompiled-project-plugin"
+                id "project-plugin"
+                id "buildsrc-plugin"
+            }
+
+            tasks.register("success") {
+                dependsOn gradle.includedBuild('$includedProject.name').task(':includedTask')
+            }
+        """
+
+        when:
+        succeeds "success"
+
+        then:
+        plugin.invokedExecutionPhaseStartedCallbackOnce(output)
```

Comment:
It would be great if we could also assert that the callback was triggered just before the top-level build tasks executed.